docs/writing-your-first-rclrs-node.md

```diff
@@ -0,0 +1,214 @@
+# Writing your first `rclrs` node
+
+This tutorial is intended to point out some of the differences of the Rust client library with the other client libraries. It assumes knowledge of Rust, and is also not intended to be an introduction to ROS 2.
+
+As a semi-realistic example, let's create a node that periodically republishes the last message it received. It's limited to only one specific message type – `std_msgs/msg/String` in this example.
+
+## Create a package
+
+You can create a package in the usual way:
+
+```console
+cargo new republisher_node && cd republisher_node
+```
+
+In the `Cargo.toml` file, add a dependency on `rclrs = "*"` and `std_msgs = "*"`.
+
+Additionally, create a new `package.xml` if you want your node to be buildable with `colcon`. See the official [tutorial on creating a ROS 2 package](https://docs.ros.org/en/rolling/Tutorials/Creating-Your-First-ROS2-Package.html) for the structure of this file. Make sure to change the build type to `ament_cargo` and the list of dependencies to the two packages mentioned above, as such:
+
+```xml
+<depend>rclrs</depend>
+<depend>std_msgs</depend>
+
+<export>
+  <build_type>ament_cargo</build_type>
+</export>
```

Comment:
I wonder how hard it would be to extend `ros2 pkg` to have a `cargo` build type that would generate all this for you.

Either way, for the ease of going through this tutorial would you mind if this block was extended to include the rest of a minimal `package.xml` for this tutorial?

```xml
<package format="3">
 <name>republisher_node</name>
 <version>0.0.0</version>
 <description>TODO: Package description</description>
 <maintainer email="user@todo.todo">user</maintainer>
 <license>TODO: License declaration</license>

 <depend>rclrs</depend>
 <depend>std_msgs</depend>

 <export>
  <build_type>ament_cargo</build_type>
 </export>
</package>
```

Reply:
Yep, good suggestion, it's not that long.